# The Adequacy of the most Commonly-Used Models in Phylogenetic Comparative Methods for Gene Expression Data Sets

This repository contains my analysis of model adequacy for a series of gene expression data sets. Links to the repositories for my analysis on data sets not contained in this repository are in the Separate Repositories category. All the gene expression data sets used in my total analysis are summarized in the table at the bottom. The directories are as follows:

### Gene Expression Data Analysis

Each of the directories below contains an Analysis.md file that summarizes the findings for that data set.

-   **cichlids**: Analysis of gene expression in 6 organs of African Cichlid species from the paper entitled [Gene expression dynamics during rapid organismal diversification in African cichlid fishes](https://www.nature.com/articles/s41559-020-01354-3)

-   **fishes:** Analysis gill tissue of a clade of fish that have partially adapted to extreme, high-sulfide environments, from the paper entitled [Ecology drives the degree of convergence in the gene expression of extremophile fishes](https://www.biorxiv.org/content/10.1101/2021.12.13.472416v1.full)

-   **Heliconius_Butterflies:** Analysis of gene expression in eye and brain of 5 species of butterfly in the Heliconius genus from the paper entitled [Drift and Directional Selection Are the Evolutionary Forces Driving Gene Expression Divergence in Eye and Brain Tissue of Heliconius Butterflies](https://academic.oup.com/genetics/article/213/2/581/5930581?login=true)

-   **Mammal_organs:** Analysis of a clade including primates and amniote out-groups of 6 different organs from the paper entitled [The evolution of gene expression levels in mammalian organs](https://www.nature.com/articles/nature10532)

-   **snakes:** Analysis of toxin expression in snakes species from the paper entitled [Toxin expression in snake venom evolves rapidly with constant shift in evolutionary rates](https://royalsocietypublishing.org/doi/full/10.1098/rspb.2020.0613#d1e1203)

### Miscellaneous 

-   **Arbutus_Exploration:** This directory contains simulation data regarding the usage of Arbutus.

    -   My initial set of simulations; where I visualized how adequate alternate models would be when the data is truly generated by an Ornstein-Uhlenbeck process, is summarized in the document entitled: [Adequacy-of-Alternate-OU-Models.md](https://github.com/pennell-lab-ubc/fiels_workbench/blob/main/Arbutus_Exploration/Adequacy-of-Alternate-OU-Models.md).

    -   My second set of simulations; where I explored how adequate single rate models were for a truly multi-rate Brownian Motion process, are summarized in the document entitled: [Analysis-of-Mulirate-Models.md](https://github.com/pennell-lab-ubc/fiels_workbench/blob/main/Arbutus_Exploration/Analysis-of-Multirate-Models.md).

-    **Data_Recreation_Task:** This directory contains early work where I attempted to recreate some data via tree simulations from the `geiger` package.

-   **debugging**: This directory was used to isolate the issues I was having with the performance of the M.sig test statistic in `arbutus` as well as S.hgt.

### Separate Repositories

-   [**comparative_expression_2017**](https://github.com/fieldima/comparative_expression_2017): This repository contains the files used to analyze the [KMRR data set](https://academic.oup.com/bib/article/18/2/205/2562739?login=false).

-   [**GeneExpression_coevolution**](https://github.com/fieldima/GeneExpression_coevolution)**:** This repository contains my analysis on gene expression in related fungi species from the paper entitled [Gene expression of functionally-related genes coevolves across fungal species: detecting coevolution of gene expression using phylogenetic comparative methods.](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-020-6761-3)

-   [**interspecific_rnaseq**](https://github.com/fieldima/interspecific_rnaseq)**:** This repository contains my analysis on vision loss in cave crayfish from the paper entitled [The Evolution of Gene Expression Underlying Vision Loss in Cave Animals.](https://academic.oup.com/mbe/article/35/8/2005/5000155?login=false)

-   [**amalgam_data**](https://github.com/fieldima/amalgam_data)**:** This repository contains my analysis on data amalgamated from 182 different data sets for 6 organs in 21 species from the paper[Amalgamated cross-species transcriptomes reveal organ-specific propensity in gene expression evolution](https://www.nature.com/articles/s41467-020-18090-8)

-   [**gene-phylogeny-pipeline**](https://github.com/pennell-lab-ubc/gene-phylogeny-pipeline)**:** This repository contains a data pipeline to generate gene family trees to reanalyze the data in GeneExpression_coevolution using the gene family phylogenies.

-   [**evogene-arbutus-comparison**](https://github.com/fieldima/evogene-arbutus-comparison)**:** This repository contains a comparative analysis of the `evogene` and `arbutus` tools in first simulated data and then real data.

### Summary Table

| Data set                    | Number of Genes | Number of Species |
|-----------------------------|-----------------|-------------------|
| cichlids                    | 32,596          | 73                |
| fishes                      | 16,740          | 20                |
| Heliconius_Butterflies      | 2,393           | 5                 |
| Mammal_organs               | 5,320           | 10                |
| snakes                      | 11              | 52                |
| comparative_expression_2017 | 8,333           | 9                 |
| GeneExpression_coevolution  | 3,556           | 18                |
| interspecific_rnaseq        | 3,560           | 14                |
| amalgam data                | 1,377           | 21                |

