# The Effect of Normalization on Model Adequacy

## Directory Contents

### Sub-directories

-   Trinity: This directory contains the Trinity-Assembled transcriptomes, the results of CD-HIT, and the results of OrthoFinder

-   Scripts: This directory contains the scripts used in this analysis; e.g., downloading sequence data via FTP, Count analysis, etc.

-   Results: This directory contains the read counts of each sample, generated by the Trinity tool and separated by species.

### Files

-   environment.yml: This is the file you will use to build the conda environment for the Snakemake workflow.

-   config.yml: This YAML file will be used to configure Snakemake.

-   Normalization_Between_Studies.csv: This file records the adequacy level of different experiments by normalization type.

## Introduction

In this directory, I will be examining the effect of count-normalization on the adequacy of different models from PCMs using the "Interspecific RNA" data set. This paper analyzed gene expression evolution in 13 crayfish species.

## Dependencies

The analyses carried out in this directory require multiple tools, most of which are contained in the conda environment described below.

The conda environment contains the Trinity Pipeline (insert reference), which itself calls Bowtie, as well as dependencies for Snakemake, Trimmomatic, OrthoFinder, and CD-HIT. The conda environment can be installed by typing the following command into terminal:

`conda env create --file environment.yml`

## Usage

Before running the Snakemake pipeline, activate the environment with the following command:

`conda activate snakemake`

Next, edit the config.yml file with the paths to the crayfish raw RNA-reads from SRA, as well as the current working directory for Snakemake. Finally, you can now run the workflow with the following command.

`snakemake -c24`

You can substitute the 24 with the maximum amount of cores you will allow Snakemake to use to run this process.

The pipeline make take some time to run, so I recommend running it in a tmux channel to prevent disconnects from disrupting the process. If the process is disrupted, you can just restart the workflow with the command above and it will carry on from where it left off.
