---
title: "Analysis"
author: "Fiel Dimayacyac"
output: github_document
---

## Summary of Data Set

The data set analyzed in this directory is taken from a study on gene expression evolution among a set of 10 species across 6 different organs. In total, they performed RNA-Seq on 5320 orthologous genes/transcripts across these species and organs. Through a hypothesis-testing framework, they showed that the rate of gene expression evolution varies across organs and species due to selective pressures. Specifically, they used hypothesis testing to determine that different lineages (or organs) had 1: Multiple evolutionary optima and 2: Had different evolutionary optima. In this case, the null hypothesis was an OU model, with the alternative hypothesis being a multi-optima OU model.

```{r include=FALSE}
#Code taken from summary of coevolution dataset, still need to edit.
#Summary stats for Arbutus results
library(tidyverse)
p_br <- readRDS("species_phylogeny/arbutus/pvals_br") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "br")
p_cb <- readRDS("species_phylogeny/arbutus/pvals_cb") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "cb")
p_ht <- readRDS("species_phylogeny/arbutus/pvals_ht") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "ht")
p_lv <- readRDS("species_phylogeny/arbutus/pvals_lv") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "lv")
p_kd <- readRDS("species_phylogeny/arbutus/pvals_kd") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "kd")
p_ts <- readRDS("species_phylogeny/arbutus/pvals_ts") %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = "ts")

p_df <- full_join(p_br, p_cb) %>% full_join(p_ht) %>% full_join(p_kd) %>% full_join(p_lv) %>% full_join(p_ts)

figure1 <- p_df %>% ggplot(aes(x = inade, y = n, fill = inade)) + geom_bar(stat = "identity") + 
  xlab("Number of inadequacies") + ylab("Number of genes") + ggtitle("Amount of genes by number of inadequacies") + facet_grid(~organ) + theme_bw()

```

```{r, echo=FALSE}
figure1

```

**Figure 1.** Number of inadequacies per gene per organ. Most genes were adequate.

Overall, each organ had a similar pattern of inadequacies after full analysis. 63% or more of the genes showed no inadequacies in all organs when fitting the best-fit model, leaving 37% of genes being inadequate when fitting their best fit models. However, interestingly none of the models being tested account for multiple rates or multiple optima, which the authors of the paper heavily rely on. This may suggest that a multi-optima method may not have been necessary, but testing multi-optima OU models would be more conclusive. Furthermore, many (but still a minority) of the genes show higher relative fit for a simple BM model over OU, suggesting that perhaps even models without optima at all may be better fitting for this data set.

## Methods

------------------------------------------------------------------------

I first conducted a relative fit analysis via AIC to compare how well each of the 3 models (BM, OU, EB) fit the data in a relative sense. Next, I performed adequacy analysis using arbutus for the best fit model as chosen above. I then conducted the same analysis for using just BM or just OU to see how it affected adequacy of the data set.

## Results

------------------------------------------------------------------------

After running through relative fit analysis, each of the organs showed similar patterns; i.e., that once again OU was the best fitting model most of the time, but BM models are the best-fitting almost as often as OU models are.

<img src="species_phylogeny/AIC/AIC_br.png" width="332"/>

<img src="species_phylogeny/arbutus/arbutus_br.png" width="313"/>

**Figure 2.** Relative fit (left) and absolute adequacy (right) of the best fit model for brain tissue. Other tissues show similar patterns.

As shown by the relative and absolute measures of adequacy above, OU models are relatively the best fit, but many data sets are best fit by BM. Overall, most inadequacies are in two test statistics: "c.var" and "s.asr". C.var accounts for rate heterogeneity, suggesting that a model that allows heterogeneous evolutionary rates may be more adequate. In the future, this could mean measuring the adequacy of a multirate BM model, or multirate OU model. S.asr accounts for trait variation relative to size of the trait. For example, if a gene is showing high expression, does it change a lot, or a little relative to size.

Next, I wanted to understand how many of the inadequacies in the data set are from C.var and S.asr respectively.

```{r, include=FALSE}
arbutus_summary <- function(organ, model){
  filepath <- paste0("species_phylogeny/arbutus/just", model, "/", model, "pvals_", organ)
  result <- readRDS(filepath) %>% select(!m.sig) %>%
    transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt),
              d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>%
    mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade), organ = organ)
  result
}
OU1 <- arbutus_summary("br", "OU")
OU2 <- arbutus_summary("cb", "OU")
OU3 <- arbutus_summary("ht", "OU")
OU4 <- arbutus_summary("kd", "OU")
OU5 <- arbutus_summary("lv", "OU")
OU6 <- arbutus_summary("ts", "OU")

OU_df <- full_join(OU1, OU2) %>% full_join(OU3) %>% full_join(OU4) %>% full_join(OU5) %>% full_join(OU6)

BM1 <- arbutus_summary("br", "BM")
BM2 <- arbutus_summary("cb", "BM")
BM3 <- arbutus_summary("ht", "BM")
BM4 <- arbutus_summary("kd", "BM")
BM5 <- arbutus_summary("lv", "BM")
BM6 <- arbutus_summary("ts", "BM")

BM_df <- full_join(BM1, BM2) %>% full_join(BM3) %>% full_join(BM4) %>% full_join(BM5) %>% full_join(BM6)

OU_df %>% ggplot(aes(x = inade, y = n, fill = inade)) + geom_bar(stat = "identity") + 
  xlab("Number of inadequacies") + ylab("Number of genes") + ggtitle("Amount of genes by number of inadequacies for OU") + facet_grid(~organ) + theme_bw()

BM_df %>% ggplot(aes(x = inade, y = n, fill = inade)) + geom_bar(stat = "identity") + 
  xlab("Number of inadequacies") + ylab("Number of genes") + ggtitle("Amount of genes by number of inadequacies for BM") + facet_grid(~organ) + theme_bw()

p_fuse <- p_df %>% mutate(model = "best")
BM_fuse <- BM_df %>% mutate(model = "BM")
fused_df <- OU_df %>% mutate(model = "OU") %>% full_join(p_fuse) %>% full_join(BM_fuse)
figure4 <- fused_df %>% filter(inade == 0) %>% ggplot(aes(x = model, y = prop, fill = model)) + geom_bar(stat = "identity") + 
  xlab("Model") + ylab("Proportion of adequate genes") + ggtitle("Proportion of adequate genes when fitting different models") + 
  facet_grid(~organ) + theme_bw() + geom_text(aes(label = paste0(round(prop*100, 1), "%")), position = position_dodge(0.9), angle = 90)
```

```{r, include=FALSE}
pvals_br <- readRDS("species_phylogeny/arbutus/pvals_br")
test <- pvals_br %>% select(!m.sig) %>% drop_na() %>%
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) 

figure3 <- test %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% 
  ggplot(aes(x = `test statistic`, y = value/sum(value), fill = `test statistic`)) + geom_col() + geom_text(aes(label = round(value/sum(value), digits = 2))) + ylab("Percent of inadequacies") + theme_bw()

#Of the 1631 inadequacies, s.asr and c.var account for 1294 of them
```

```{r, echo=FALSE}
figure3
```

**Figure 3.** Proportion of inadequacies belonging to each test statistic. C.var was responsible for most of the inadequacies.

As shown by the plot above, c.var and s.asr account for almost 80% of the inadequacies. This means that finding a model that addresses both of these factors is imperative.

Finally, due to BM data sets showing a relatively high count of genes in which it is the best fit model, I wanted to compare the absolute adequacy of fitting purely BM, OU, or best fit models and see how important the evolutionary optima and alpha parameters are for this dataset.

```{r}
figure4
```

**Figure 4.** Proportion of adequate genes from fitting each model to the data, by organ.

As shown by the figure above, BM models lose large amounts of adequacy, even though they represent the best fit model for a large portion of data sets. Interestingly, OU models do not lose much adequacy compared to best-fit models, suggesting that even when the BM model is the best-fit, the OU model is good enough most of the time. Overall this shows the importance of the optima and constraint parameters of the OU model.

## Conclusion

------------------------------------------------------------------------

Analysis of this data set reveals that none of the 3 used models (BM, OU, or EB) were fully adequate for the data for any of the organs, but that OU model is mostly good. An improvement over the OU model would then be an OU model with multiple rates, and perhaps multiple optima, as shown by the inadequacies in specifically c.var and s.asr.
