---
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Analysis of Extremophile Fishes Data Adequacy

------------------------------------------------------------------------

## Introduction

The data set is taken from a [paper](https://www.biorxiv.org/content/10.1101/2021.12.13.472416v1.full) studying transcriptomes in gill tissue of a genera of fishes, where 10 of the species have colonized pools high in sulfides. The paper primarily leverages the EVE model (which is an extension of OU) to study gene expression shifts between taxa that live in sulfide-rich waters and those that do not. They found that the degree of "convergence" in gene expression was higher not in species that were closely related, but those that live in similar environments. This suggests that convergent evolution of gene expression is occurring. This analysis aims to uncover if the phylogenetic models at hand are adequate for the data set.

## Summary Analysis

```{r include=FALSE}
library(tidyverse)
pval_files <- list.files("arbutus/pvals/")

readfile <- function(filename){
  res <- readRDS(paste0("arbutus/pvals/", filename))
  res
}

readfile2 <- function(filename){
  res <- readRDS(paste0("arbutus/multirate/pvals/", filename))
  res
}

pvals_total <- map_df(pval_files, readfile)

p_df <- pvals_total %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade))

figure1 <- p_df %>% ggplot(aes(x = inade, y = n)) + geom_bar(stat = "identity", fill = "#E5B80B") + geom_text(aes(label = round(prop, digits = 2)), size = 6) +
  xlab("Number of inadequacies") + ylab("Number of genes") + theme_classic() + theme(strip.text = element_text(size = 15), legend.position = "none", axis.title = element_text(size = 15), axis.text = element_text(size = 15), title = element_text(size = 20), plot.title = element_text(hjust = 0.5)) + ggtitle("W/o BMS") + scale_fill_brewer(palette = "Set2")


pvals_multi <- list.files("arbutus/multirate/pvals/") %>% grep("^p", value = TRUE, x = .) %>% map_df(readfile2)
pvals_motmot <- list.files("arbutus/multirate/pvals/") %>% grep("^m", value = TRUE, x = .) %>% map_df(readfile2)

na_total <- pvals_total %>% select(s.hgt) %>% filter(is.na(s.hgt)) %>% summarise(`test statistic` = "s.hgt NAs", value = n(), models = "w/o BMS")
na_multi <- pvals_multi %>% select(s.hgt) %>% filter(is.na(s.hgt)) %>% summarise(`test statistic` = "s.hgt NAs", value = n(), models = "w/ BMS")

p_dfb <- pvals_multi %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade))

figure1b <- p_dfb %>% ggplot(aes(x = inade, y = n)) + geom_bar(stat = "identity", fill = "#E5B80B") + geom_text(aes(label = round(prop, digits = 2)), size = 6) +
  xlab("Number of inadequacies") + ylab("Number of genes") + theme_classic() + theme(strip.text = element_text(size = 15), legend.position = "none", axis.title = element_text(size = 15), axis.text = element_text(size = 15), title = element_text(size = 20), plot.title = element_text(hjust = 0.5)) + ggtitle("W/ BMS") + scale_fill_brewer(palette = "Set2")

p_dfc <- pvals_motmot %>% select(!m.sig) %>% transmute(c.less = c.var <= 0.05, sv.less = s.var <= 0.05, sa.less = s.asr <= 0.05, sh.less = s.hgt <= 0.05 & !is.na(s.hgt), d.less = d.cdf <= 0.05) %>% transmute(inade = c.less + sv.less + sa.less + sh.less + d.less) %>% count(inade) %>% mutate(prop = n/sum(n)) %>% mutate(inade = as.character(inade))

figure1c <- p_dfc %>% ggplot(aes(x = inade, y = n)) + geom_bar(stat = "identity", fill = "#E5B80B") + geom_text(aes(label = round(prop, digits = 2)), size = 6) +
  xlab("Number of inadequacies") + ylab("Number of genes") + theme_classic() + theme(strip.text = element_text(size = 15), legend.position = "none", axis.title = element_text(size = 15), axis.text = element_text(size = 15), title = element_text(size = 20), plot.title = element_text(hjust = 0.5)) + ggtitle("W/ Motmot") + scale_fill_brewer(palette = "Set2")

pvals_total2 <- pvals_total %>% mutate(Type = "noBMS")
pvals_multi2 <- pvals_multi %>% mutate(Type = "BMS")
pvals_motmot2 <- pvals_motmot %>% mutate(Type = "Motmot")

pvals_all <- full_join(pvals_total2, pvals_multi2) %>% full_join(pvals_motmot2) %>% pivot_longer(cols = -c(Type), names_to = "Tstat", values_to = "Pvals") %>% filter(!Tstat == "m.sig")

pvals_all %>% ggplot(aes(x = Pvals)) + geom_histogram() + facet_grid(rows = vars(Type), cols = vars(Tstat))
```

```{r, echo=FALSE}
figure1
figure1b
figure1c
```

**Figure 1: Using the best-fit model (chosen by AIC) shows a low adequacy for the data set. A)** Without the addition of a multi-rate Brownian Motion model shows less than half of genes being fully adequate. **B)** The addition of a multi-rate Brownian Motion model increases the adequacy of the data set to just over half.

Overall, the data set shows inadequacies in multiple test-statistics. 20 fish species are included, with 16740 genes overall.

## Results

### Initial Arbutus Analysis

<img src="arbutus/AIC/AIC_1.png" width="271"/>

<img src="arbutus/plots/arbutus_all.png" width="407"/>

**Figure 2. Relative fit (left) and absolute fit (right) of the data without inclusion of a BMS model.** Overall, a BM model fits the data the best in a relative sense, and in an absolute sense the best-fit model shows problems in all test statistics except for d.cdf.

After initial analysis, I hypothesized that adding a multi-rate model to the analysis would increase the fit of the data, due to inadequacies in s.asr, s.var, s,hgt, and c.var which were shown to have increased values under 0.05 when the true model is BMS and a single-rate model is fit in my other analysis (Analysis of Multirate models in the Arbutus Exploration folder). In addition, the paper that the data stems from studied rate-shifts between different lineages depending on environment.

For the multi-rate Brownian Motion analysis, regimes have to be set that signify differences in mutation rate. Due to the paper's analysis of fish gene expression differences due to environmental factors (living in a high-sulfur habitat or not), I set selection regimes to follow the habitat patterns of the fish. The tree below illustrates this.

```{r, echo=FALSE}

library(ape)
tree <- read.tree("data/recodedTreeNamed.tre") %>% chronoMPL()
plot(tree)
```

**Figure 3. Phylogenetic tree of species included in this analysis. Species with "S" signify that this species lives in a sulfur-rich environment, with "NS" signifying they do not.** Tips labeled "S" were considered one selective regime, with "NS" being another.

I then performed relative and absolute fit analysis in line with the previous analysis displayed in Figure 2 but allowing a BMS model to be included.

<img src="arbutus/multirate/AIC/AIC_1.png" width="271"/>

<img src="arbutus/multirate/plots/arbutus_all.png" width="407"/>

**Figure 4. Relative fit (left) and absolute fit (right) of the data with inclusion of a BMS model.** Overall, the BMS model fits the data the best in a relative sense, and in an absolute sense the best-fit model shows problems in all test statistics except for d.cdf. However, compared to Figure 2, it seems that inadequacies in the coefficient of variation (c.var) have decreased.

Overall it seems that the new, multi-rate model fits the data better in a relative sense, and seems to cause changes in inadequacies. Specifically, it causes less issues with c.var. I then wanted to analyze what changes were causing the overall adequacy to increase when using a BMS model; what was the BMS model adding to the absolute fit?

```{r, include=FALSE}
test <- pvals_total %>% select(!m.sig) %>%
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/o BMS")

test2 <- pvals_multi %>% 
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/ BMS")

test3 <- pvals_motmot %>% 
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/ Motmot")

tables <- full_join(test, test2) %>% full_join(test3)
tables$models <- as.factor(tables$models) %>% fct_relevel("w/o BMS", "w/ BMS", "w/ Motmot")

tables_wBMS <- tables %>% filter(models == "w/ BMS") %>% mutate(value = value/nrow(pvals_multi), value = round(value, digits = 3))
tables_woBMS <- tables %>% filter(models == "w/o BMS") %>% mutate(value = value/nrow(pvals_total), value = round(value, digits = 3))
tables_motmot <- tables %>% filter(models == "w/ Motmot") %>% mutate(value = value/nrow(pvals_total), value = round(value, digits = 3))

tables_tot <- full_join(tables_wBMS, tables_woBMS) %>% full_join(tables_motmot)

figure5 <- tables_tot %>% 
  ggplot(aes(x = `test statistic`, y = value, fill = `test statistic`)) + geom_col() +
  geom_text(aes(label = round(value, digits = 2)), size = 6) + facet_wrap(vars(models)) + theme_bw() + theme(strip.text = element_text(size = 15), legend.position = "none", axis.title = element_text(size = 15), axis.text = element_text(size = 15), panel.grid = element_blank()) + ylab("Proportion of Genes") + xlab("Test Statistic") + scale_fill_brewer(palette = "Set1")

```

```{r, echo=FALSE}
figure5
```

```{r, include=FALSE}
readfile <- function(filename){
  res <- readRDS(paste0("arbutus/compressed/pvals/", filename))
  res
}

readfile2 <- function(filename){
  res <- readRDS(paste0("arbutus/multirate/compressed/pvals/", filename))
  res
}

pvals_total <- list.files("arbutus/compressed/pvals/") %>% map_df(readfile)
pvals_multi <- list.files("arbutus/multirate/compressed/pvals/") %>% grep("^p", value = TRUE, x = .) %>% map_df(readfile2)
pvals_motmot <- list.files("arbutus/multirate/compressed/pvals/") %>% grep("^m", value = TRUE, x = .) %>% map_df(readfile2)

test <- pvals_total %>% select(!m.sig) %>%
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/o BMS")

test2 <- pvals_multi %>% 
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/ BMS")

test3 <- pvals_motmot %>% 
  transmute(c.var = c.var <= 0.05, s.var = s.var <= 0.05,
              s.asr = s.asr <= 0.05, s.hgt = s.hgt <= 0.05 & !is.na(s.hgt),
              d.cdf = d.cdf <= 0.05) %>% drop_na() %>% summarise(across(.fns = ~sum(.x, na.rm = TRUE))) %>% pivot_longer(cols = everything(), names_to = "test statistic") %>% mutate(models = "w/ Motmot")

tables <- full_join(test, test2) %>% full_join(test3)
tables$models <- as.factor(tables$models) %>% fct_relevel("w/o BMS", "w/ BMS", "w/ Motmot")

tables_wBMS <- tables %>% filter(models == "w/ BMS") %>% mutate(value = value/nrow(pvals_multi), value = round(value, digits = 3))
tables_woBMS <- tables %>% filter(models == "w/o BMS") %>% mutate(value = value/nrow(pvals_total), value = round(value, digits = 3))
tables_motmot <- tables %>% filter(models == "w/ Motmot") %>% mutate(value = value/nrow(pvals_total), value = round(value, digits = 3))

tables_tot <- full_join(tables_wBMS, tables_woBMS) %>% full_join(tables_motmot)

figure6 <- tables_tot %>% 
  ggplot(aes(x = `test statistic`, y = value, fill = `test statistic`)) + geom_col() +
  geom_text(aes(label = round(value, digits = 2)), size = 6) + facet_wrap(vars(models)) + theme_bw() + theme(strip.text = element_text(size = 15), legend.position = "none", axis.title = element_text(size = 15), axis.text = element_text(size = 15), panel.grid = element_blank()) + ylab("Proportion of Genes") + xlab("Test Statistic") + scale_fill_brewer(palette = "Set1")

```

```{r, echo=FALSE}
figure6
```

**Figure 5. Number of inadequacies in the data set by test statistic.** Addition of BMS model decreased the number of inadequacies due to c.var and s.asr drastically. Slight increases in s.hgt and s.var, with d.cdf staying approximately equal.

Addition of the BMS model decreased the inadequacies of the data (when using the best-fit model overall) by decreasing issues with C.var and S.asr; which are both related to rate variation. Interestingly, it seemed to increase issues with S.var and S.hgt. However, the increase in inadequacies for s.hgt can be attributed to a decrease in the amount of NA values, as shown by the figure above; the total value of s.hgt counts plus s.hgt NA values stays consistent between the two. Still, overall adequacy of the models for the data set increased with this addition. Relative fit of BMS by AIC also seems to indicate that the BMS model was not too complex as to break the assumptions of AIC, suggesting that BMS models are better overall than the single-rate models used previously.
